src/helpers/api.coffee

```diff
@@ -33,42 +33,100 @@ makeLocalRequest = (requestConfig) ->
     url_parts = ["#{uri}.json", params]
   else
     url_parts = ["#{uri}/#{requestConfig.method}.json", params]
+    requestConfig.mockMethod = requestConfig.method
     requestConfig.method = 'GET'
 
+
   requestConfig.url = _.compact(url_parts).join('?')
 
 makeLocalResponse = (response) ->
   payload = if response.config.data then JSON.parse(response.config.data) else {}
   response.data = _.extend({}, response.data, payload)
 
-setUpXHRInterceptors = ->
-  axios.interceptors.request.use (requestConfig) ->
-    AppActions.queRequest(requestConfig)
 
-    makeLocalRequest(requestConfig) if IS_LOCAL
+interceptors =
+  queRequest: (requestConfig) ->
+    AppActions.queRequest(requestConfig)
+    requestConfig
 
+  makeLocalRequest: (requestConfig) ->
+    makeLocalRequest(requestConfig)
     requestConfig
 
-  axios.interceptors.response.use (response) ->
-    {status, statusText, config} = response
-    AppActions.setServerSuccess(status, statusText, config)
+  setResponse: (response) ->
+    {status, statusText, config, headers} = response
 
-    makeLocalResponse(response) if IS_LOCAL
+    AppActions.setServerSuccess(status, statusText, config)
+    setNow(headers)
 
     response
 
-  , (errorResponse) ->
-
+  handleError: (errorResponse) ->
     if _.isError(errorResponse)
       errorResponse =
         status: 1
         statusText: "#{errorResponse.name}: #{errorResponse.message}"
-    else
-      {status, statusText, config} = errorResponse
-      AppActions.setServerError(status, statusText, config)
+    else if errorResponse.headers
+      setNow(errorResponse.headers)
+
+    Promise.reject(_.extend({handled: false}, errorResponse))
+
+  makeLocalResponse: (response) ->
+    makeLocalResponse(response)
+    response
+
+  handleMalformedRequest: (errorResponse) ->
+    if errorResponse.status is 400
+      CurrentUserActions.logout()
+      errorResponse.handled = true
 
     Promise.reject(errorResponse)
 
+  handleNotFound: (errorResponse) ->
+    if errorResponse.status is 404
+      errorResponse.statusText = 'ERROR_NOTFOUND'
+
+    Promise.reject(errorResponse)
+
+  handleLocalErrors: (errorResponse) ->
+    {status, statusText, config} = errorResponse
+    {method, mockMethod} = config
+
+    mockMethods = ['PUT', 'PATCH']
+
+    # Hack for local testing, fake successful PUT and PATCH
```

Comment:
I kinda think it's better not to have this now that `makeRequestLocal` handles this, or this should return as a different error that makes it clear where the fake `json` should be made to handle the fake request.